config/configmapprovider/expand.go

```diff
@@ -86,6 +87,10 @@ func expandEnv(s string) string {
 		if str == "$" {
 			return "$"
 		}
-		return os.Getenv(str)
+		pair := strings.Split(str, ":")
```

Comment:
On `${ENV:a:b}` if `ENV` is unset it will be replaced by `a` instead of `a:b`.  You need to use [`SplitN`](https://pkg.go.dev/strings#SplitN) instead with `n` equals 1, to split only by the first separator `:`

Reply:
Good catch. Now using `SplitN`